youtube_dl/extractor/yahoo.py

```diff
@@ -556,3 +560,153 @@ def _real_extract(self, url):
                 'https://gyao.yahoo.co.jp/player/%s/' % video_id.replace(':', '/'),
                 YahooGyaOPlayerIE.ie_key(), video_id))
         return self.playlist_result(entries, program_id)
+
+
+class YahooJapanNewsIE(InfoExtractor):
+    IE_NAME = 'yahoo:japannews'
+    IE_DESC = 'Yahoo! Japan News'
+    _VALID_URL = r'https?://(?P<host>(?:news|headlines)\.yahoo\.co\.jp)[^\d]*(?P<id>\d[\d-]*\d)?'
+    _TESTS = [{
+            'url': 'https://headlines.yahoo.co.jp/videonews/ann?a=20190716-00000071-ann-int',
+            'info_dict': {
+                'id': '1736242',
+                'ext': 'mp4',
+                'title': 'ムン大統領が対日批判を強化“現金化”効果は？（テレビ朝日系（ANN）） - Yahoo!ニュース',
+                'description': '韓国の元徴用工らを巡る裁判の原告が弁護士が差し押さえた三菱重工業の資産を売却して - Yahoo!ニュース(テレビ朝日系（ANN）)',
+                'thumbnail': r're:^https?://.*\.[a-zA-Z\d]{3,4}$',
+            }, 'params': {'skip_download': True},
+        }, {
+            'url': 'https://headlines.yahoo.co.jp/videonews/',
+            'only_matching': True,
+        }, {
+            'url': 'https://news.yahoo.co.jp',
+            'only_matching': True,
+        }, {
+            'url': 'https://news.yahoo.co.jp/byline/hashimotojunji/20190628-00131977/',
+            'only_matching': True,
+        }, {
+            'url': 'https://news.yahoo.co.jp/feature/1356',
+            'only_matching': True
+        }]
+
```

Comment:
Add `_GEO_COUNTRIES` to bypass geo restrictions. Example URL https://news.yahoo.co.jp/pickup/6330725. Add as test.

Reply:
Okay, done and done.